File: cloud-collect-core/src/main/java/com/windea/demo/cloudcollect/core/domain/view/UsernamePasswordLoginView.java
package com.windea.demo.cloudcollect.core.domain.view;

import lombok.Data;

/**
 * 用户名&密码登录视图。
 */
@Data
public class UsernamePasswordLoginView implements UserLoginView {
	private static final long serialVersionUID = 8037232370825960415L;

	private final String username;

	private final String password;
}

File: cloud-collect-core/src/main/kotlin/com/windea/demo/cloudcollect/core/domain/view/UsernamePasswordLoginView.kt
package com.windea.demo.cloudcollect.core.domain.view

import com.windea.demo.cloudcollect.core.validation.annotation.*
import javax.validation.constraints.*

/**用户名&密码登录视图。 */
data class UsernamePasswordLoginView(
	@field:NotEmpty(message = "{validation.User.username.NotEmpty}")
	@field:Username(message = "{validation.User.username.ValidUsername}")
	val username: String,
	
	@field:NotEmpty(message = "{validation.User.password.NotEmpty}")
	@field:Password(message = "{validation.User.password.ValidPassword}")
	val password: String
) : LoginView
